eclipse/src/saros/editor/internal/EditorAPI.java

```diff
@@ -268,27 +272,275 @@ public static IResource getEditorResource(IEditorPart editorPart) {
+       * this call also uses IDocument.getLineOfOffset(...) but simply drops the thrown exception.
+       */
+      startLine = document.getLineOfOffset(offset);
+      startLineOffset = document.getLineOffset(startLine);
+
+    } catch (BadLocationException e) {
+      LOG.warn(
+          "Could not calculate the start line or start line offset for the given start offset "
+              + offset
+              + " in the document "
+              + document,
+          e);
+
+      return TextSelection.EMPTY_SELECTION;
+    }
+
+    int startInLineOffset = offset - startLineOffset;
+    TextPosition startPosition = new TextPosition(startLine, startInLineOffset);
+
+    // Skip second part of calculation if no selection is present
+    if (endOffset == offset) {
+      return new TextSelection(startPosition, startPosition);
+    }
+
+    int endLine;
+    int endLineOffset;
+
+    try {
+      /*
+       * Explicitly does not use textSelection.getEndLine() to obtain the start line number as this
+       * call also uses IDocument.getLineOfOffset(...) but simply drops the thrown exception.
+       */
+      endLine = document.getLineOfOffset(endOffset);
+
+      // Skip line offset calculation if end is in same line as start
+      if (endLine == startLine) {
+        endLineOffset = startLineOffset;
+
+      } else {
+        endLineOffset = document.getLineOffset(endLine);
+      }
+
+    } catch (BadLocationException e) {
+      LOG.warn(
+          "Could not calculate the end line or end line offset for the given end offset "
+              + endOffset
+              + " in the document "
+              + document,
+          e);
+
+      return TextSelection.EMPTY_SELECTION;
+    }
+
+    int endInLineOffset = endOffset - endLineOffset;
+    TextPosition endPosition = new TextPosition(endLine, endInLineOffset);
+
+    return new TextSelection(startPosition, endPosition);
+  }
+
+  /**
+   * Calculates the text based offsets in the given editor part for the given text selection.
+   *
+   * <p>The given editor part must not be <code>null</code> and the given text selection must not be
+   * <code>null</code> and must not be empty.
+   *
+   * @param editorPart the local editor part
+   * @param selection the text selection
+   * @return the absolute offsets in the given editor part for the given text selection or {@link
+   *     org.eclipse.jface.text.TextSelection#emptySelection()} if the given editor is not a text
+   *     editor, the editor does not have a valid document provider, a valid IDocument could not be
+   *     obtained form the document provider, or the line offsets could not be calculated
+   * @throws NullPointerException if the given editor part or selection is <code>null</code>
+   * @throws IllegalArgumentException if the given text selection is empty
+   */
+  public static ITextSelection calculateOffsets(IEditorPart editorPart, TextSelection selection) {
```

Comment:
I guess both methods can easily be JUnit tested ?

Reply:
No, not really...We would need a working eclipse backing or would have to mock the necessary methods (like the requests to `IDocument`, `SelectionProvider`, etc).

Reply:
I do not think that ``` public static TextSelection calculateSelection(IDocument document, int offset, int length)``` is hard to test. Even this method is not that hard to test. You need to setup 3 mocks where 2 mocks do not really do anything.

Reply:
I added tests for both methods for both the Eclipse and IntelliJ implementation.

It is way overbuild in my opinion, but it is a test suite.